luda-editor/client/src/pages/graphic_asset_manage_page/mod.rs

```diff
@@ -0,0 +1,213 @@
+    fn render(self, ctx: &RenderCtx) -> RenderDone {
+        const TOP_BAR_HEIGHT: Px = px(48.0);
+        const SIDE_BAR_WIDTH: Px = px(192.0);
+
+        let Self { wh, project_id } = self;
+
+        let project_id = ctx.track_eq(&project_id);
+        let (tab, _set_tab) = ctx.atom_init(&TAB_ATOM, || Tab::Image);
+        let (_, _) = ctx.atom_init(&IMAGES_ATOM, Vec::new);
+        let (_, _) = ctx.atom_init(&CG_FILES_ATOM, Vec::new);
+        let (selected_asset, _) = ctx.atom_init(&SELECTED_ASSET_ATOM, || None);
+
+        ctx.effect("Fetch graphic assets every project_id changes", || {
+            start_fetch_graphic_assets(*project_id);
+        });
+
+        let on_viewer_close = || {
+            SELECTED_ASSET_ATOM.set(None);
+        };
+        ctx.compose(|ctx| {
+            let Some(selected_asset) = selected_asset.deref() else {
+                return;
+            };
+
+            match selected_asset {
+                SelectedAsset::Image(image) => {
+                    ctx.add_with_key(
+                        "image_viewer",
+                        ImageViewer {
+                            wh,
+                            image,
+                            project_id: *project_id,
+                            on_close: &on_viewer_close,
+                        },
+                    );
+                }
+                SelectedAsset::Cg(cg_file) => {
+                    ctx.add_with_key(
+                        "cg_viewer",
+                        CgViewer {
+                            wh,
+                            cg_file,
+                            project_id: *project_id,
+                            on_close: &on_viewer_close,
+                        },
+                    );
+                }
+            }
+        });
+
+        ctx.compose(|ctx| {
+            vertical([
+                fixed(TOP_BAR_HEIGHT, |wh, ctx| {
+                    ctx.add(TopBar {
+                        wh: Wh::new(wh.width, TOP_BAR_HEIGHT),
+                        project_id: *project_id,
+                    });
+                }),
+                ratio(1, |wh, ctx| {
+                    horizontal([
+                        fixed(SIDE_BAR_WIDTH, |wh, ctx| {
+                            ctx.add(SideBar { wh });
+                        }),
+                        ratio(1, |wh, ctx| {
+                            match *tab {
+                                Tab::Image => ctx.add(ImageList {
+                                    wh,
+                                    project_id: *project_id,
+                                }),
+                                Tab::Cg => ctx.add(CgList {
+                                    wh,
+                                    project_id: *project_id,
+                                }),
+                            };
+                        }),
+                    ])(wh, ctx);
+                }),
+            ])(wh, ctx)
```

Comment:
Nah

Reply:
Please check all of this cases in this PR.

Reply:
Fixed with 3279e15